src/FG2_fmt_plug.c

```diff
@@ -0,0 +1,281 @@
+/*
+ * Fortigate (FortiOS) SHA256 Password cracker
+ *
+ * This software is Copyright (c) 2018 Nicolas B. <mrtchuss at gmail.com>,
+ * and it is hereby released to the general public under the following terms:
+ * Redistribution and use in source and binary forms, with or without modification, are permitted.
+ *
+ * Based on work on Fortigate AK1 format by  Mat G. <mat.jtr at gmail.com>, (Copyright (c) 2012)
+ *
+ * Passwords are located in "config system admin" part of the configuration file :
+ *
+ * config system admin
+ *     edit "<username>"
+ *        set password ENC SH2+n+OiTlautpJQF1NBcMU2H1otOvlAz9g5wzxwHw+4il3Aiixv9oPtKnRgCc=
+ *
+ * Password is : SH2|base64encode(salt|hashed_password)
+ * where hashed_password is SHA256(salt|password|fortinet_magic)
+ *
+ * salt is 12 bytes long
+ * hashed_password is 32 bytes long (SHA256)
+ * encoded password is 63 bytes long (3 bytes for SH2 and 60 bytes of base64encode(salt|hashed_password))
+ *
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_FG2;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_FG2);
+#else
+
+#include <string.h>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#include "common.h"
+#include "formats.h"
+#include "misc.h"
+#include "sha.h"
+#include "base64_convert.h"
+#include "memdbg.h"
+
+#define FORMAT_LABEL		"Fortigate256"
+#define FORMAT_NAME             "FortiOS256"
+#define ALGORITHM_NAME		"SHA256 32/" ARCH_BITS_STR
+
+#define BENCHMARK_COMMENT	""
+#define BENCHMARK_LENGTH	0
+
+#define PLAINTEXT_LENGTH	44
+#define CIPHERTEXT_LENGTH	60
+#define HASH_LENGTH             CIPHERTEXT_LENGTH + 3
+
+#define BINARY_SIZE             32
+#define BINARY_ALIGN		4
+#define SALT_SIZE               12
+#define SALT_ALIGN		4
+
+#define FORTINET_MAGIC          "\xa3\x88\xba\x2e\x42\x4c\xb0\x4a\x53\x79\x30\xc1\x31\x07\xcc\x3f\xa1\x32\x90\x29\xa9\x81\x5b\x70"
+#define FORTINET_MAGIC_LENGTH   24
+
+#define MIN_KEYS_PER_CRYPT		1
+#define MAX_KEYS_PER_CRYPT		512
+
+#ifdef __MIC__
+#ifndef OMP_SCALE
+#define OMP_SCALE               16
+#endif
+#else
+#ifndef OMP_SCALE
+#define OMP_SCALE               2 // Tuned w/ MKPC for core i7
```

Comment:
Is this actually tuned or was this copy-paste?

Reply:
This is copy-paste from FGT_fmt_plug.c :); i will remove this